core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -186,8 +189,13 @@ public void injectDependencies(GroupManager groupManager, ScatteredVersionManage
          } else { // not primary owner
             CompletableFuture<Map<Address, Response>> rpcFuture =
                rpcManager.invokeRemotelyAsync(info.writeOwners(), command, defaultSyncOptions);
-            return asyncValue(rpcFuture.thenApply(responseMap ->
-                  handleWritePrimaryResponse(ctx, command, responseMap)));
+            return asyncValue(rpcFuture.thenApply(responseMap -> {
+               try {
+                  return handleWritePrimaryResponse(ctx, command, responseMap);
+               } catch (Throwable t) {
+                  throw CompletableFutures.asCompletionException(t);
```

Comment:
With `asyncValue(rpcFuture).thenApply((rCtx, rCommand, responseMap) -> ...)` you wouldn't need the `try..catch` (but you would need a cast for the `responseMap`).